### DISCUSSION TITLE
Setting neighborhood methods in Gen-Ens-Prod config file

### DISCUSSION THREAD

**Discussion by michelleharrold** [dtcenter/METplus#1317] [Configuration] [answered]
Hi There,

I am setting up some tests using MET's Gen-Ens-Prod tool. In the default MET configuration file, there are several sections that indicate they may be set separately in each field entry (e.g., the regrid and ensemble product output types). The 'nbrhd_prob' and 'nmep_smooth' sections do not contain that disclaimer. Is it possible to set them specific to each field entry as well?

Thanks!
Michelle Harrold

**Comment by jprestop** [comment]
Hi @michelleharrold.  

I confirmed with @JohnHalleyGotway that at this time, because of where nbrhd_prob and nmep_smooth are currently defined, they can only be set once for each run of Gen-Ens-Prod and not customized separately for each field entry.  If your project needs them to be set separately, the code could be enhanced to do so. But we'd need to write up an issue to capture that enhancement and figure out when to do it and how to pay for it.

@georgemccabe added that if your project needs to define the value differently for each field and we are not able to make the changes to the MET tool soon, it may be possible to set that up in a METplus use case. You would need to create separate instances in GenEnsProd in the process list with instance IDs and process a single field in each instance with the different neighborhood values set for each. See: https://metplus.readthedocs.io/en/latest/Users_Guide/systemconfiguration.html#process-list

Please let us know how you'd like to proceed.

**Comment by michelleharrold** [comment]
Hi @jprestop,

Thanks for looking into this!

The project doesn't necessarily "need" this functionality, but it is certainly a "want" -- there is some customization with variables/ thresholds/neighborhood methods that would be nice, but it is not a showstopper. For example, for light precipitation, we might want to use the NEP method, but for moderate-to-heavy precipitation, we would want to use NMEP. The current consequence is we will either have bloated output files with unnecessary variable/threshold/neighborhood method permutations, or we will have a number of METplus conf files for all the permutations we want to support.

I am tagging @jwolff-ncar since she is the project lead and has a bigger picture idea of prioritizing paying for this.

Thanks,
Michelle

**Comment by jwolff-ncar** [comment, marked as answer]
Thank you for add this question @michelleharrold and for your response @jprestop. I agree with Michelle that this falls into the category of being a nice enhancement that would allow for more targeted methods to be applied for specific fields and would also remove unnecessary output. I think writing up an issue for this would be great so it is on the list. If you assess the resources required and the timeline we can discuss priority level and funding options.
Thanks!

**Reply by jprestop** [reply]
Thank you @michelleharrold and @jwolff-ncar.  I have created an https://github.com/dtcenter/MET/issues/1997 [issue](https://github.com/dtcenter/MET/issues/1997) for this task, but it is a little bare.  

@TaraJensen, @JohnHalleyGotway I left the Time Estimate empty for the developer to assess.  I also left the Relevant Deadlines, Funding Source, and Requestor blank.  It also needs project assignment and a Milestone set.  Could you please take a look at set as you see fit?
